Modules/main.c

```diff
@@ -250,6 +358,16 @@ pymain_run_command(wchar_t *command)
         goto error;
     }
 
+    // Only perform auto-dedent if the string starts with a newline
+    if (strncmp(PyBytes_AsString(bytes), "\n", 1) == 0) {
+        PyObject *new_bytes = _utf_8_bytes_dedent(bytes);
+        if (new_bytes == NULL) {
```

Comment:
Maybe fall back to the original bytes in this case

Reply:
I think that if this fails, we ought to just error. The only case I can see it failing is if `PyBytes_FromStringAndSize` fails to allocate memory and returns NULL.